### PR TITLE
More kibana_system changes to support Fleet transform install

### PR DESCRIPTION
**Pull request by droberts195**
Changes to support elastic/kibana#142920.

Transform destination indices now have a version number appended to their names, so the patterns used to set up the index privileges need to be adjusted accordingly.

(Marked `>non-issue` as the release note for #91499 covers it.)

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java
Comment (kevinlog) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:771]: I think this should be `/metrics-.*&~(metrics-endpoint\\.metadata_current_default*` in order to be backwards compatible with legacy indices in the Endpoint package.

This also has me thinking that another privilege would be needed here.  I believe we need `PutMappingAction` for the index aliases that are created.  One index that isn't accounted for is, for instance `.metrics-endpoint.metadata_united_default.all`.  I think we should also add `.metrics-endpoint.metadata_united_default*` to account for that.  Let me know your thoughts.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java
Comment (kevinlog) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:824]: thank you for adding these!

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java
Comment (droberts195) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:771]: This line is different to the others because it's a regex, so `.*` in this line is effectively like `*` in the others.

> I believe we need `PutMappingAction` for the index aliases that are created.

I think that's `IndicesAliasesAction` which is granted on line 827.

@qn895 do we ever put mappings directly on the transform destination indices? Or do we always use an index template? If we ever directly set mappings directly then more changes are needed in this file.

> I think we should also add `.metrics-endpoint.metadata_united_default*` to account for that.

I don't think it's needed because it begins with a dot. This line is saying the permissions on line 778 should be applied to `metrics-*` indices _unless_ they also match `metrics-endpoint.metadata_current_default*`. So `.metrics-endpoint.metadata_current_default*` and `.metrics-endpoint.metadata_united_default*` don't match because they start with a dot.

Then all of `metrics-endpoint.metadata_current_default*`, `.metrics-endpoint.metadata_current_default*` and `.metrics-endpoint.metadata_united_default*` instead get the permissions on line 827.

Comment (kevinlog) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:771]: > > I believe we need `PutMappingAction` for the index aliases that are created.

> I think that's `IndicesAliasesAction` which is granted on line 827.

> @qn895 do we ever put mappings directly on the transform destination indices? Or do we always use an index template? If we ever directly set mappings directly then more changes are needed in this file.

Got it thanks for the clarification.  I defer to @qn895 for more info.

> I don't think it's needed because it begins with a dot. This line is saying the permissions on line 778 should be applied to `metrics-*` indices _unless_ they also match `metrics-endpoint.metadata_current_default*`. So `.metrics-endpoint.metadata_current_default*` and `.metrics-endpoint.metadata_united_default*` don't match because they start with a dot.

> Then all of `metrics-endpoint.metadata_current_default*`, `.metrics-endpoint.metadata_current_default*` and `.metrics-endpoint.metadata_united_default*` instead get the permissions on line 827.

Got it makes sense, thanks for the clarification

**Review by kevinlog — APPROVED** [review]

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java
Comment (qn895) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:771]: > @qn895 do we ever put mappings directly on the transform destination indices? Or do we always use an index template? If we ever directly set mappings directly then more changes are needed in this file.

We do not put mappings directly on the transform destination indices, just the index & component templates. The only thing we set directly when creating these indices is the alias.

**Review by qn895 — APPROVED** [review]

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java
Comment (droberts195) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:771]: > We do not put mappings directly on the transform destination indices, just the index & component templates.

OK great. I think the permissions we've got now should work then.

**Comment by slobodanadamovic** [comment]
Pinging @elastic/kibana-security for awareness

**Review by slobodanadamovic — APPROVED** [review]
LGTM from `es-security` perspective, still need approval from @elastic/kibana-security

**Review by azasypkin — APPROVED** [review]
Looks good from the Kibana Platform Security perspective too, thanks for the ping!

**Pull request merged**
